engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -906,6 +907,21 @@ - (void)viewWillAppear:(BOOL)animated {
   [super viewWillAppear:animated];
 }
 
+- (void)viewIsAppearing:(BOOL)animated {
+  TRACE_EVENT0("flutter", "viewIsAppearing");
+  if (self.engine.viewController == self) {
+    // The earliest the view can access the scene is in viewIsAppearing.
```

Comment:
It looks like the `UIWindowScene` first becomes accessible to the view in `[UIView didMoveToWindow]` (when establishing the view hierarchy. `UIWindow`'s initializer takes a `UIWindowScene`), and `viewIsAppearing` is called by the layout process so it's a bit later than that:
```
  * frame #0: 0x000000018545a010 UIKitCore`-[UIViewController viewIsAppearing:]
    frame #1: 0x0000000185449ab8 UIKitCore`-[UIViewController __viewIsAppearing:skipWindowCheck:] + 188
    frame #2: 0x0000000186142cd4 UIKitCore`-[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1072
```
Not sure if that makes a difference tho.

Reply:
The [windowScene](https://developer.apple.com/documentation/uikit/uiwindow/windowscene) property is not readonly I guess that means a `UIWindow` can be moved to a different scene while the app is running? I think that is something that can happen in an add-to-app scenario? Also the viewController associated with the engine can be changed too (https://api.flutter.dev/ios-embedder/interface_flutter_engine.html#a9247723288f07f7f7cc6d85fb701fbdc), so the engine pointers associated with a scene may also need to be updated on those events?

Reply:
Given that the engine instance associated with a FlutterViewController can change, would it be better to change the addViewController API (currently private) to addEngine, also do we need a `deleteEngine` method if, say, two FlutterViewControllers with different `UIWindowScene`s swap their engines?